component/loki/write/write.go

```diff
@@ -32,6 +34,27 @@ type Arguments struct {
 	Endpoints      []EndpointOptions `river:"endpoint,block,optional"`
 	ExternalLabels map[string]string `river:"external_labels,attr,optional"`
 	MaxStreams     int               `river:"max_streams,attr,optional"`
+	WAL            WalArguments      `river:"wal,block,optional"`
+}
+
+// WalArguments holds the settings for configuring the Write-Ahead Log (WAL) used
+// by the underlying remote write client.
+type WalArguments struct {
+	Enabled          bool          `river:"enabled,attr,optional"`
+	MaxSegmentAge    time.Duration `river:"max_segment_age,attr,optional"`
+	MinReadFrequency time.Duration `river:"min_read_frequency,attr,optional"`
+	MaxReadFrequency time.Duration `river:"max_read_frequency,attr,optional"`
+}
+
+func (wa *WalArguments) SetToDefault() {
```

Comment:
nit: Do we also want to implement the Validator interface to eg. make sure that `min_read_frequency < max_read_frequency < max_segment_age`?